### ISSUE TITLE
[Bug]: There is no flushed-L0 segments but we have been doing L0Compaction tasks

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#31191]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240311-a99143dd-amd64
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):   pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. test: concurrent insert + delete + flush + search
2. After the number of flushed-L0 segments becomes 0, the number of L0CompactionDone keeps increasing?
![img_v3_028t_de69a66f-e814-4ec0-897c-f90a946a6d7g](https://github.com/milvus-io/milvus/assets/27288593/87564106-6106-4d55-97b3-1e0f651d5def)

### Milvus Log

- https://argo-workflows.zilliz.cc/archived-workflows/qa/cefa4c7e-7c1d-4054-ad7d-6d587a15e31e?nodeId=level-zero-last-1a-2411773871
- [metrics of level-zero-upsert-op-1-7800](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=P1809F7CD0C75ACF3&var-cluster=&var-namespace=qa-milvus&var-instance=level-zero-upsert-op-1-7800&var-collection=All&var-app_name=milvus&from=1710141329520&to=1710211573423)
- pods:
```
level-zero-upsert-op-1-7800-etcd-0                                1/1     Running            0                  19h
level-zero-upsert-op-1-7800-etcd-1                                1/1     Running            0                  19h
level-zero-upsert-op-1-7800-etcd-2                                1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-datanode-679f9bd59c-bwnl9      1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-datanode-679f9bd59c-xnt6m      1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-indexnode-f8569d8-ntjtc        1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-indexnode-f8569d8-qnl7q        1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-mixcoord-6c6c5956f4-xnw4t      1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-proxy-5dfb86ddb4-gp6jh         1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-querynode-0-64bc54fd99-9p8lf   1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-querynode-0-64bc54fd99-m4959   1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-querynode-0-64bc54fd99-pfsv9   1/1     Running            0                  19h
level-zero-upsert-op-1-7800-milvus-querynode-0-64bc54fd99-rs4nl   1/1     Running            0                  19h
level-zero-upsert-op-1-7800-minio-0                               1/1     Running            0                  19h
level-zero-upsert-op-1-7800-minio-1                               1/1     Running            0                  19h
level-zero-upsert-op-1-7800-minio-2                               1/1     Running            0                  19h
level-zero-upsert-op-1-7800-minio-3                               1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-bookie-0                       1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-bookie-1                       1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-bookie-2                       1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-bookie-init-6gldl              0/1     Completed          0                  19h
level-zero-upsert-op-1-7800-pulsar-broker-0                       1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-proxy-0                        1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-pulsar-init-crwr4              0/1     Completed          0                  19h
level-zero-upsert-op-1-7800-pulsar-recovery-0                     1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-zookeeper-0                    1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-zookeeper-1                    1/1     Running            0                  19h
level-zero-upsert-op-1-7800-pulsar-zookeeper-2                    1/1     Running            0                  19h
```

**Comment by XuanYang-cn** [comment]
/assign @ThreadDao
Fixed, please help verify

/unassign

**Comment by ThreadDao** [comment]
fixed `milvus-io-master-773c64e-20240314`
